[5.7] Argon and Bcrypt hasher classes do not verify hashing algorithm

ioluas opened the pull request:
Add additional check on hashing algorithm when checking a hashed value. An exception is thrown if algorithm does not match the one intended in the hasher.

Closes #24162.

jhaboush committed:
Fix hashing classes not verifying algorithm

Add additional check on hashing algorithm when checking a hashed
value. An exception is thrown if algorithm does not match the
one intended in the hasher.

tillkruss commented:
The tests are failing because PHP 7.2 does not implement the `PASSWORD_AGRON2I` algorithm in it's default build.

We need to either update the Travis CI config, @ioluas maybe you can switch to using the `algoName` instead of `also`?

jhaboush committed:
Fix hashing classes not verifying algorithm

Modified test on password info to check algoName instead.

ioluas commented:
I just pushed a commit with updated code. That seems to fail too. Maybe I should skip the tests if `PASSWORD_ARGON2I` isn't defined?

tillkruss commented:
I see, yeah, if you can get Argon working on Travis CI, please do.

jhaboush committed:
Fix hashing classes not verifying algorithm

    Now skipping `ArgonHasher` test on platforms where PHP isn't built with
    Argon support

ioluas commented:
Hope that's sorted now. Thanks for your help.

a-komarev commented on src/Illuminate/Hashing/ArgonHasher.php:
```diff
@@ -81,13 +82,19 @@ public function make($value, array $options = [])
      * @param  array  $options
      * @return bool
+     *
+     * @throws Exception
```
Use FQN name here? `@throws \Exception`

a-komarev commented on src/Illuminate/Hashing/BcryptHasher.php:
```diff
@@ -65,13 +66,19 @@ public function make($value, array $options = [])
      * @param  array   $options
      * @return bool
+     *
+     * @throws Exception
```
Use FQN name here? `@throws \Exception`
